sdk/spring/spring-cloud-azure-feature-management/src/main/java/com/azure/spring/cloud/feature/management/FeatureManager.java

```diff
@@ -73,45 +91,178 @@ public Mono<Boolean> isEnabledAsync(String feature) {
+
+        if (feature == null) {
+            throw new FeatureManagementException("The Feature " + featureName + " can not be found.");
+        }
+
+        validateVariant(feature, featureName);
+
+        // Disabled?
+        if (!feature.getEvaluate() && StringUtils.hasText(feature.getAllocation().getDefautlWhenDisabled())) {
+            return variantAssignment.getVariant(feature.getVariants().values(),
+                feature.getAllocation().getDefautlWhenDisabled()).single();
+        } else if (!feature.getEvaluate()) {
+            return Mono.justOrEmpty(null);
+        } else if (feature.getEnabledFor().size() == 0) {
+            return variantAssignment.assignVariant(feature.getAllocation(), feature.getVariants().values());
+        }
+
+        List<Mono<Boolean>> results = new ArrayList<>();
+
+        for (FeatureFilterEvaluationContext featureFilter : feature.getEnabledFor().values()) {
+            if (StringUtils.hasText(featureFilter.getName())) {
+                results.add(isFeatureOn(featureFilter, feature.getKey(), featureContext));
+            }
+        }
+
+        Mono<Boolean> isEnabled;
+
+        // All Filters must be true
+        if (ALL_REQUIREMENT_TYPE.equals(feature.getRequirementType())) {
+            isEnabled = Flux.merge(results).reduce((a, b) -> a && b).single();
+        } else {
+            // Any Filter must be true
+            isEnabled = Flux.merge(results).reduce((a, b) -> a || b).single();
+        }
+
+        return isEnabled.flatMap(enabled -> {
+            if (!enabled && StringUtils.hasText(feature.getAllocation().getDefautlWhenDisabled())) {
+                return variantAssignment.getVariant(feature.getVariants().values(),
+                    feature.getAllocation().getDefautlWhenDisabled()).single();
+            } else if (!enabled) {
+                return Mono.justOrEmpty(null);
+            }
+
+            return variantAssignment.assignVariant(feature.getAllocation(), feature.getVariants().values());
+        });
+
+    }
+
+    private void validateVariant(Feature feature, String featureName) {
+        if (feature.getVariants() == null || feature.getVariants().size() == 0) {
+            throw new FeatureManagementException("The " + feature.getKey() + " has no assigned Variants.");
+        }
+
+        for (VariantReference variant : feature.getVariants().values()) {
+            if (!StringUtils.hasText(variant.getName())) {
+                throw new FeatureManagementException("Variant needs a name");
+            }
+
+            if (variant.getConfigurationValue() == null && variant.getConfigurationReference() == null) {
+                throw new FeatureManagementException(
+                    "The " + feature.getKey() + " neededs a Configuration Value or Configuration Reference.");
+            }
+            if (variant.getConfigurationValue() != null && variant.getConfigurationReference() != null) {
```

Comment:
In .NET, if both are present then we prioritize the configuration value and ignore the reference. We just need to make that clear in the docs.